### PR TITLE
waf: require python 3.8.0 or later

### PR DESCRIPTION
**Pull request by tridge**
for 4.5.x we no longer support py2

### CODE REVIEW

**Comment by Ryanf55** [comment]
LGTM. The error is descriptive. 3.8 matches what's in Ubuntu 20.04. 
```
ryan@ryan-B650:~/Development/ardu_ws/src/ardupilot$ python2 waf configure
Setting top to                           : /home/ryan/Development/ardu_ws/src/ardupilot 
Setting out to                           : /home/ryan/Development/ardu_ws/src/ardupilot/build 
Autoconfiguration                        : enabled 
Checking for program 'python'            : /usr/bin/python2 
Checking for python version >= 3.8.0     : 2.7.18 
The python version is too old, expecting (3, 8, 0)
(complete log in /home/ryan/Development/ardu_ws/src/ardupilot/build/config.log)
```

File: wscript [hunk 1]
```diff
@@ -456,6 +456,10 @@ def configure(cfg):
     if cfg.options.num_aux_imus > 0:
         cfg.define('INS_AUX_INSTANCES', cfg.options.num_aux_imus)
 
+    # require python 3.8.x or later
+    cfg.load('python')
+    cfg.check_python_version(minver=(3,8,0))
```
Comment (peterbarker) [inline wscript:461]:
```suggestion
    cfg.check_python_version(minver=(3,6,0))
```

Let Ubuntu 18.04 continue to function

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: wscript (hunk 1 above)
Comment (khancyr) [reply wscript:461]: yep, 3.8 don't bring much functionnality against 3.6, so using 3.6 is good

Comment (peterbarker) [reply wscript:475]: > yep, 3.8 don't bring much functionnality against 3.6, so using 3.6 is good

Discussed this with tridge.

bionic came with 3.8 and it's almost defunct.

Python 3.6 is *also* out of support.

**Review by khancyr — APPROVED** [review]
LGTM

**Pull request merged**